src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/settings/ProjectReferenceFileController.java

```diff
@@ -93,12 +94,7 @@ public String getProjectReferenceFilesPage(final Model model, final Principal pr
 			map.put("label", file.getLabel());
 			map.put("createdDate", file.getCreatedDate());
 			Path path = file.getFile();
-			try {
-				map.put("size", Files.size(path));
-			} catch (IOException e) {
-				logger.error("Cannot find the size of file " + file.getLabel());
-				map.put("size", messageSource.getMessage("projects.reference-file.not-found", new Object[] {}, locale));
-			}
+			map.put("size", iridaFileStorageService.getFileSize(path));
```

Comment:
Should we maybe add some of these file methods into the model classes?  That way we don't need to wire the `IridaFileStorageService` into some places it maybe doesn't belong (like this class).

Some examples I can see: `getFileSize`, `getFileInputStream`, maybe some of the other things that are like "read" operations on the file.

Reply:
Yup that's a good idea!

Reply:
Updated in 8477bb95ad7d2dd6a2e12591e61c4870bba42f7f